nca/FWRules/DotGraph.py

```diff
@@ -74,9 +76,11 @@ def add_node(self, subgraph, name, node_type, label):
         param label: node label
         """
         label = [tok.strip() for tok in label if tok != '']
+        title = subgraph if self.do_not_subgraph else ''
+        subgraph = '' if self.do_not_subgraph else subgraph
         if subgraph not in self.subgraphs:
             self.subgraphs[subgraph] = self.Subgraph(subgraph)
-        node = self.Node(name, node_type, label)
+        node = self.Node(name, node_type, label, title)
```

Comment:
lines 81-82 should not check if `subgraph` is empty  string and ignore it in that case?

Reply:
no,
there is  also a subgraph without name . 
and all nodes w/o a subgraph are at this subgraph
it is also later given as a parameter to _subgraph_to_str()